src/main/java/edu/umich/its/cpm/MigrationController.java

```diff
@@ -1,32 +1,196 @@
+        	log.info("Wrong user id or password. Cannot login to CTools " + env.getProperty("ctools.direct.url"));
+        }
+        else
+        {
+        	// get the session id
+            sessionId = response.getBody();
+            log.info("successfully logged in as user " + env.getProperty("username") + " with sessionId = " + sessionId);
+            
+        	// 2. become the user
+        	// retrieve from the configuration file for now; Shoudl get from the REMOTE_USER setting after CoSign integration
+        	restTemplate = new RestTemplate();
+        	// the url should be in the format of "https://server/direct/session/SESSION_ID.json"
+        	requestUrl = env.getProperty("ctools.direct.url") + "session/becomeuser/" + env.getProperty("become.username")+".json?_sessionId=" + sessionId;
+        	log.info(requestUrl);
+
+        	try
+        	{
+        		String resultString = restTemplate.getForObject(requestUrl, String.class, sessionId); 
+        		log.info(resultString);
+        	}
+        	catch (RestClientException e)
+        	{
+        		log.error(requestUrl + e.getMessage());
+        		
+        		// nullify sessionId if become user call is not successful
+        		sessionId = null;
+        	}
+        }
+        return sessionId;
+    }
+
+    /**
+     * get all migration records
+     * @return
+     */
     @RequestMapping("/migrations")
     public String migrations() {
-    	Gson gson = new Gson();
-    	return gson.toJson(repository.findAll());
+    	return JSONObject.valueToString(repository.findAll());
     }
     
+    /**
+     * get a specific migration record
+     * @param migration_id
+     * @return
+     */
+    @RequestMapping("/migrations/{migration_id}")
+    public String migrations(@PathVariable("migration_id") Integer migration_id) {
+    	return JSONObject.valueToString(repository.findOne(migration_id));
+    }
+    
+    /**
+     * found all migrated records (where the migration record have "end" field value
+     * @return
+     */
     @RequestMapping("/migrated")
     public String migrated() {
-    	Gson gson = new Gson();
-    	return gson.toJson(repository.find());
+    	return JSONObject.valueToString(repository.findMigrated());
+    }
+    
+    /**
+     * insert a new record of Migration
+     * @param request
+     */
+    @RequestMapping(value="/migration", method = RequestMethod.POST)
+    public void migrated(HttpServletRequest request) {
+	    Map<String, String[]> parameterMap = request.getParameterMap();
+	    Migration m = new Migration(parameterMap.get("site_id")[0],
+	    		parameterMap.get("site_name")[0],
+	    		parameterMap.get("tool_id")[0],
+	    		parameterMap.get("tool_name")[0],
+	    		parameterMap.get("migrated_by")[0],
+	    		new java.sql.Timestamp(System.currentTimeMillis()), /* start time is now*/
+				null, /* no end time */
+				parameterMap.get("destination_type")[0],
+				null);
+	    repository.save(m);
```

Comment:
Commit is included in the save method. According to the doc, it is not clear what type of exception will be thrown for save method, but I have added a try-catch block for general Exception anyway.

http://docs.spring.io/spring-data/commons/docs/current/api/org/springframework/data/repository/CrudRepository.html#save-S-